### ISSUE TITLE
Incorrect type inferred from loader for `any` in 1.17.0

### ISSUE THREAD

**Issue by iamkd** [remix-run/remix#6559]
### What version of Remix are you using?

1.17.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

```tsx
export async function loader() {
  const result: any = "test"

  return json(result)
}

export default function Index() {
  const result = useLoaderData<typeof loader>(); // const result: { [x: string]: any; }
  return <div>{result}</div>
}
```

### Expected Behavior

The type of `result` in the component should be `any`.

### Actual Behavior

In the component, `result` gets a type of `{
    [x: string]: any;
}`. 

This happens for any data structure passed to `json`. if deep inside the object/array there is a field with type `any`, this field's type will be replaced with `{ [x: string]: any; }`.

**Comment by Marcdj-02** [comment]
Experiencing the same. It's also the case for Prisma.JsonValue which results in: 
![image](https://github.com/remix-run/remix/assets/72502650/9e150991-465f-4db1-9cd9-48774770249d)

**Comment by MichaelDeBoey** [comment]
Fixed by #6582, which will be released in the next release
